background.js

```diff
@@ -0,0 +1,6 @@
+function onBeforeRequest(data) {
+    console.error('top of onBeforeRequest')
+    // figure out how and what to block here
+}
+
+chrome.webRequest.onBeforeRequest.addListener(onBeforeRequest);
```

Comment:
The structure looks good, but as Casey and I discovered earlier, Chrome will complain about only one parameter being passed to `addListener`. The filters object is also required, it turns out. You'll want to carefully explore the 'Register event listeners' section of the webRequest API doc (https://developer.chrome.com/extensions/webRequest) to find details on what shape that filter object (which, as you'll discover, should be an instance of `RequestFilter`) should have. The same section has a link to URL patterns docs which will explain how to specify a value for the `urls` property on the filters object. You want to set it up so that every http and https request gets forward to our handler.